torch/ao/quantization/fx/_decomposed.py

```diff
@@ -824,25 +747,16 @@ def dequantize_per_token_meta(
     return torch.empty_like(input, dtype=output_dtype)
 
 
-quantized_decomposed_lib.define(
-    "quantize_per_channel_group(Tensor input, Tensor scales, Tensor zero_points, int quant_min, "
-    "int quant_max, ScalarType dtype, int group_size) -> Tensor"
-)
-
-
-# TODO: dtype is ignored for now
-@impl(
-    quantized_decomposed_lib, "quantize_per_channel_group", "CompositeExplicitAutograd"
-)
+@custom_op(f"{ns}::quantize_per_channel_group", mutates_args=())
```

Comment:
Seems like `mutates_args` is mostly empty (100% of the case in this file). Did we decide to make this a required arg or can we make it optional?

Reply:
Also, have you considered letting the user set an explicit flag that says "get the op name from the function name" to skip the duplicate naming? So if the user sets this to true then they're acknowledging the risk of other decorators messing the op name up. Maybe something like:
```
# turns into `quantized_decomposed::quantize_per_channel_group`
@custom_op(ns, append_func_name_to_op_name=True)
def quantize_per_channel_group(...)
```
Do you think this idea makes sense or it's still too risky?

Reply:
> Seems like mutates_args is mostly empty (100% of the case in this file). Did we decide to make this a required arg or can we make it optional?

Yes, we decided to make this required. If the user does not tell us correctly what args are being mutated, this will lead to undefined behavior. If it is not required, our experience with the previous custom ops API is that people will copy-paste existing custom ops and never specify it.

> Also, have you considered letting the user set an explicit flag that says "get the op name from the function name" to skip the duplicate naming? So if the user sets this to true then they're acknowledging the risk of other decorators messing the op name up. Maybe something like:

The explicit name seemed easier to grok. But also something else is that with overload names (e.g. "quantize_decomposed::quantize_for_tensor.tensor") there is no way to infer this from the function's name.